process/block/shardblock.go

```diff
@@ -172,11 +171,16 @@ func (sp *shardProcessor) ProcessBlock(
 	log.Debug("total txs in pool",
 		"num txs", numTxWithDst,
 	)
+	// TODO: remove if start of epoch block needs to be validated by the new epoch nodes
```

Comment:
Theoretically, we definitely want the new Nodes to validate the new block. While unlikely in any case, a collusion of some Nodes to ruin a Shard's morning is made more difficult if this block is processed by freshly arrived Nodes. The new Nodes have been in this Shard at most one Epoch (in waiting), while the Nodes of the old Epoch might have been here far longer (even forever, but with very low probability).

Reply:
Yes, this was the intention for the final solution, to have the new epoch block validated by the new nodes, otherwise it would be difficult to get out of situations like consensus stuck due to many offline nodes.
This here is more of a workaround before the changes to make that possible are done.

Reply:
We could either leave a `TODO` here or make a JIRA issue.